front/components/data_source/RequestDataSourcesModal.tsx

```diff
@@ -0,0 +1,199 @@
+import { Button, DropdownMenu, Modal, TextArea } from "@dust-tt/sparkle";
+import type { WorkspaceType } from "@dust-tt/types";
+import React, { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import logger from "@app/logger/logger";
+import type { DataSourceIntegration } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+type RequestDataSourceProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  dataSourceIntegrations: DataSourceIntegration[];
+  currentUserEmail: string;
+  owner: WorkspaceType;
+};
+
+async function sendRequestDataSourceEmail({
+  email,
+  emailMessage,
+  dataSourceName,
+  emailRequester,
+  owner,
+}: {
+  email: string;
+  emailMessage: string;
+  dataSourceName: string;
+  emailRequester?: string;
+  owner: WorkspaceType;
+}) {
+  const res = await fetch(`/api/w/${owner.sId}/data_sources/request-email`, {
```

Comment:
Let's simply do request-access as it might not be an email anymore in the future.